### PR TITLE
feat(azure): customize entra and platform logs (SSPROD-43735)

### PR DESCRIPTION
**Pull request by SKosier**

### CODE REVIEW

File: modules/integrations/event-hub/main.tf [hunk 1]
```diff
@@ -121,151 +121,36 @@ resource "azurerm_role_assignment" "sysdig_data_receiver" {
...
+    for_each = var.enabled_platform_logs
+    content {
+      category = enabled_log.value
+    }
   }
 }
 
 resource "azurerm_monitor_aad_diagnostic_setting" "sysdig_entra_diagnostic_setting" {
-  count = var.enable_entra ? 1 : 0
+  count = length(var.enabled_entra_logs) > 0 ? 1 : 0
```
Comment (legobrick) [inline modules/integrations/event-hub/main.tf:140]:
```suggestion
  count = var.enable_entra && length(var.enabled_entra_logs) > 0 ? 1 : 0
```

File: modules/integrations/event-hub/variables.tf [hunk 2]
```diff
@@ -113,4 +113,16 @@ variable "enable_entra" {
...
+variable "enabled_platform_logs" {
+  description = "List of platform logs to enable. Options are: 'Administrative', 'Policy', 'Security'."
+  type        = list(string)
+  default     = ["Administrative", "Security", "Policy"]
+}
+
+variable "enabled_entra_logs" {
+  description = "List of Entra logs to enable"
+  type        = list(string)
+  default     = []
```
Comment (legobrick) [inline modules/integrations/event-hub/variables.tf:127]:
```suggestion
  default     = ["AuditLogs","SignInLogs","NonInteractiveUserSignInLogs","ServicePrincipalSignInLogs","ManagedIdentitySignInLogs","ProvisioningLogs","ADFSSignInLogs","RiskyUsers","UserRiskEvents","NetworkAccessTrafficLogs","RiskyServicePrincipals","ServicePrincipalRiskEvents","EnrichedOffice365AuditLogs","MicrosoftGraphActivityLogs","RemoteNetworkHealthLogs"]
```

**Review by legobrick — CHANGES_REQUESTED** [review]
We need to make this backwards compatible, so the default setup should enable everything. We should also continue supporting enable_entra variable for the time being, for whomever has a onboarding setup generated after that was introduced

**Review by legobrick — CHANGES_REQUESTED** [review]
Please adapt the README to add the new variables
Also mark `enabled_entra_logs` as deprecated

**Review by jacalvo — APPROVED** [review]
I can confirm this is working, tested pointing to the local module on `skosier/modular-onboarding/entra-logs` branch with this:
```
  enabled_entra_logs       = ["SignInLogs","NonInteractiveUserSignInLogs","ServicePrincipalSignInLogs","ManagedIdentitySignInLogs","ProvisioningLogs","ADFSSignInLogs","RiskyUsers","UserRiskEvents","NetworkAccessTrafficLogs","RiskyServicePrincipals","ServicePrincipalRiskEvents","EnrichedOffice365AuditLogs","MicrosoftGraphActivityLogs"]
  enabled_platform_logs    = ["Administrative","Policy"]
```
And they are created as expected:
image
image

File: modules/integrations/event-hub/README.md [hunk 3]
```diff
@@ -59,29 +59,30 @@ No modules.
...
+| <a name="input_consumer_group_name"></a> [consumer\_group\_name](#input\_consumer\_group\_name)                                          | Name of the consumer group to be created                                                                                                 | `string`       | `"sysdig-consumer-group"` | no |
+| <a name="input_diagnostic_settings_name"></a> [diagnostic\_settings\_name](#input\_diagnostic\_settings\_name)                           | Name of the diagnostic settings to be created                                                                                            | `string`       | `"sysdig-diagnostic-settings"` | no |
+| <a name="input_enable_entra"></a> [enable\_entra](#input\_enable\_entra)                                                                 | (Deprecated) Used to enable or disable Entra logs, defaults to true.                                                                             | `bool`         | `true` | no |
```
Comment (legobrick) [inline modules/integrations/event-hub/README.md:67]:
```suggestion
| <a name="input_enable_entra"></a> [enable\_entra](#input\_enable\_entra)                                                                 | (Deprecated, see [enabled_entra_logs](#input\_enabled\_entra\_logs)) Used to enable or disable Entra logs, defaults to true.                                                                             | `bool`         | `true` | no |
```

**Review by legobrick — CHANGES_REQUESTED** [review]

**Review by legobrick — APPROVED** [review]

**Pull request merged**